repository/release.go

```diff
@@ -149,7 +152,35 @@ func (r *ReleaseRepository) ListReleasesForProject(ctx context.Context, projectI
 		return nil, err
 	}
 
-	return model.ToSvcReleases(rls), nil
+	return r.toSvcReleases(rls)
+}
+
+func (r *ReleaseRepository) toSvcRelease(rls model.Release) (svcmodel.Release, error) {
+	var tagURL url.URL
+	var err error
+
+	if rls.GithubOwnerSlug.Valid && rls.GithubRepoSlug.Valid {
+		tagURL, err = r.githubURLGenerator.GenerateGitTagURL(rls.GithubOwnerSlug.String, rls.GithubRepoSlug.String, rls.GitTagName)
+		if err != nil {
+			return svcmodel.Release{}, fmt.Errorf("failed to generate tag URL: %w", err)
+		}
+	}
+
+	return model.ToSvcRelease(rls, tagURL), nil
+}
+
+func (r *ReleaseRepository) toSvcReleases(releases []model.Release) ([]svcmodel.Release, error) {
+	rls := make([]svcmodel.Release, 0, len(releases))
+	for _, release := range releases {
+		svcRls, err := r.toSvcRelease(release)
+		if err != nil {
+			return nil, err
+		}
+
+		rls = append(rls, svcRls)
+	}
+
+	return rls, nil
```

Comment:
That doesn't feel smooth. And it's a bit confusing (two `toSvcRelease()` methods).
I'd keep the URL generation in the methods where it is needed. Or in the worst case wrap it into a helper here.

And do we need to check the validity?
In case repo slug and owner slug are empty, I'd just generate an empty URL.

Reply:
I declare empty URL first

```
var tagURL url.URL 
```

Then, if the GitHub owner and repository slugs for the project are set, I generate the real URL for the tag. Otherwise, I return the empty URL.

However, I dislike the idea of generating a URL when the GitHub repository is not set. This can result in URLs like:

```
https://github.com///releases/tag/{tagName}
```

I think there is an improvement for future refactoring. If I have git tag for release, I shouldn't be worry that project repo is not set. User cannot remove repo connection (once project is connected to repository, repository can be only changed). So practically you wont end up in the state where you have git tag but project is not connected to the repository. 

But I think database model should reflect this behaviour as well because now as you can see, I need to handle the case.

----

Regarding `toSvcRelease`/`toSvcReleases`, I will try to move it to helper methods.

Reply:
>However, I dislike the idea of generating a URL even when the GitHub repository is not set. This can result in URLs like:

Yes. But my point is, I'd consider moving that logic ("if repo is empty, then return empty URL") to the `GithubClient.GenerateGitTagURLForRelease` method.
But it's really minor now, after the recent change when it's nicely wrapped into an explicit helper. 👌 

(One argument though - if you were to write a test for `GithubClient.GenerateGitTagURLForRelease`, wouldn't you also test these corner cases when the inputs are empty strings? And wouldn't you implement it, i.e. return empty URL, already on that level?)

Reply:
>(One argument though - if you were to write a test for GithubClient.GenerateGitTagURLForRelease, wouldn't you also test these corner cases when the inputs are empty strings? And wouldn't you implement it, i.e. return empty URL, already on that level?)

Good point 👍. I will merge the current PR and refactor it in a separate PR.